vignettes/Guidelines_Contribution_Dataset-formatting.Rmd

```diff
@@ -0,0 +1,235 @@
+* Generate the data objects by knitting the R markdown script.
+* Make sure the data objects are well formatted and consistent with the other datasets.
+* Check all the new package metadata and documentation.
+* Upload the data objects to `AWS S3` and announce the upload to Bioconductor Hubs.
+* Download the data objects from `ExperimentHub` and check the format again.
+* Update the `NEWS`, `DESCRIPTION` (add new contributor, version bump) and `README.md` (if needed) files.
+* Build and check the package, make sure it passes all R and Bioconductor checks.
+* Push to GitHub and check that the `imcdatasets` package can be installed from there.
+* Test the package functionality in R.
+* Once everything works, approve the pull request and merge with the master branch.
+
+Contributors will be recognized by having their names added to the DESCRIPTION file of the `imcdatasets` package.
+
+
+
+# **Dataset format**
+
+The `imcdatasets` package is meant to provide quick and easy access to published and curated IMC datasets. Each dataset consists of three data objects that can be retrieved individually:
+
+* __Single cell data__ in the form of a [SingleCellExperiment](https://bioconductor.org/packages/SingleCellExperiment) object.
+* __Multichannel images__ formatted into a [CytoImageList](https://bioconductor.org/packages/CytoImageList) object.
+* __Cell segmentation masks__ formatted into a [CytoImageList](https://bioconductor.org/packages/CytoImageList) object.
+
+The three data objects can be mapped using unique `image_name` values contained in the metadata of each object.
+
+For consistency across datasets, the guidelines below must be followed when creating a new dataset.
+
+
+## Single cell data
+
+Single cell data should be formatted into a [SingleCellExperiment](https://bioconductor.org/packages/SingleCellExperiment) object named `sce` that contains the following slots:
+
+- `colData`: observations metadata.
+- `rowData`: marker metadata.
+- `assays`: marker expression levels per cell.
+- `colPairs` *(optional)*: neighborhood information.
+
+### colData
+
+The `colData` entry of the `SingleCellExperiment` object is a `DataFrame` that contains observation metadata; i.e., cells, slides, tissue, patients, .... All column names must have a prefix that indicates the level of observation (e.g. `cell_`, `slide_` , `tissue_`, `patient_`, `tumor_`).
+
+The following columns are required:
+
+- `image_name`: unique image (ablated ROI) name. Should map to the `image_name` column in the metadata of the `images` and `masks` objects.
+- `cell_number`: integer representing cell numbers. Should map to the values of cell segmentation masks.
+- `cell_id`: a unique cell identifier defined as {`image_name` `_` `cell_number`} (e.g., `E02_188`).
+- `cell_x` and `cell_y`: position of the cell centroid on the image. These columns are used as `SpatialCoords` when converting to a [SpatialExperiment](https://bioconductor.org/packages/SpatialExperiment) object.
```

Comment:
As discussed offline, we'll keep `cell_x` and `cell_y` to avoid confusion, for instance for `image_x` and `image_y`. Users will have to add specify the `coords` argument when using `imcRtools`.